src/Mod/Part/App/TopoShapeExpansion.cpp

```diff
@@ -4460,161 +4460,142 @@ TopoShape& TopoShape::makeElementPrismUntil(
         op = Part::OpCodes::Prism;
     }
 
-    BRepFeat_MakePrism PrismMaker;
+    TopoShape usedUptoshape(uptoshape);
 
-    // don't remove limits of concave face
-    if (checkLimits && __uptoface.shapeType(true) == TopAbs_FACE) {
-        Base::Vector3d vCog;
-        profile.getCenterOfGravity(vCog);
-        gp_Pnt pCog(vCog.x, vCog.y, vCog.z);
-        checkLimits = !Part::Tools::isConcave(TopoDS::Face(__uptoface.getShape()), pCog, direction);
+    TopoShape usedBase(base);
+    if (usedBase.isNull()) {
+        Mode = PrismMode::None;
+        usedBase = profile;
     }
 
-    TopoShape _uptoface(__uptoface);
-    if (checkLimits && _uptoface.shapeType(true) == TopAbs_FACE
-        && !BRep_Tool::NaturalRestriction(TopoDS::Face(_uptoface.getShape()))) {
-        // When using the face with BRepFeat_MakePrism::Perform(const TopoDS_Shape& Until)
-        // then the algorithm expects that the 'NaturalRestriction' flag is set in order
-        // to work as expected.
-        BRep_Builder builder;
-        _uptoface = _uptoface.makeElementCopy();
-        builder.NaturalRestriction(TopoDS::Face(_uptoface.getShape()), Standard_True);
-    }
+    if (checkLimits && uptoshape.shapeType(true) == TopAbs_FACE) {
+        // disregard the supplied base,
+        // and use up to face to extrude our own base. Later on, use the
```

Comment:
should this be `uptoface` ?

Reply:
I've renamed the argument, `uptoface` is declared at line 4478.